dimod/generators/satisfiability.py

```diff
@@ -31,17 +31,24 @@
 
 def _kmcsat_interactions(num_variables: int, k: int, num_clauses: int,
                          *,
+                         is_planted: bool = False,
```

Comment:
probably  `plant_solution: bool = False` or even just `plant: bool = False`  is fine, here and elsewhere.

Reply:
You mean 'planted_solution' for compatibility with other generators in the module. I'll use that.

Reply:
There is a distinction between `planted_solution: Mapping[Variable, int]` and `plant_solution: bool`. In the former case the user is providing it, in the latter it's an instruction to the function.

Reply:
To make the generators more consistent would make sense to have a plant: bool and planted_solution:  Optional[Mapping[Variable, int]], for both fcl and kcsat(nae3sat/2in4SAT). plant defaulted to True in the former and False in the latter.
The way planting is dealt with in frustrated_loop() is a bit inefficient and over complicated. Would be clearer and more efficient to plant 1 and apply an SRT at the end if planted_solution is not None. Might make this change unless there is an objection.

Reply:
That's fine. I would even go so far as:
1) Make this one `plant_solution: bool`
2) Add fcl `plant_solution: bool`
3) Throw a deprecation warning if `planted_solution` is provided, and suggest using SRTs (though keep the functionality for now for backwards compatibility)

Reply:
That's probably the outcome I prefer. I think we should encourage people to use SRTs for planted instances - pushing the randomness into the solver rather than the instance class (where it is a superficial feature). Some solvers are SRT dependent and others not, we should encourage users to notice this.